src/main/java/com/sonymobile/jenkins/plugins/mq/mqnotifier/RunListenerImpl.java

```diff
@@ -61,67 +63,74 @@ public RunListenerImpl() {
 
     @Override
     public void onStarted(Run r, TaskListener listener) {
-        if (r instanceof AbstractBuild) {
-            AbstractBuild<?, ?> build = (AbstractBuild<?, ?>)r;
-            List<String> causes = new LinkedList<String>();
-            for (Object o : build.getCauses()) {
-                causes.add(o.getClass().getSimpleName());
-            }
+        List<String> causes = new LinkedList<String>();
+        for (Object o : r.getCauses()) {
```

Comment:
getCauses() gives a List<Cause>. Why iterating it on object types? If we want to add any other info from the cause it will add unnecessary type casting.